lib/web_ui/lib/src/engine/text/ruler.dart

```diff
@@ -168,6 +187,44 @@ class ParagraphGeometricStyle {
   }
 }
 
+/// Contains all styles that have an effect on the height of text.
+///
+/// This is useful as a cache key for [TextHeightRuler]. It's more efficient
+/// than using the entire [ParagraphGeometricStyle] as a cache key.
+class TextHeightStyle {
+  TextHeightStyle({
+    required this.fontFamily,
+    required this.fontSize,
+    required this.height,
+    required this.fontFeatures,
+  });
+
+  final String fontFamily;
+  final double fontSize;
+  final double? height;
+  final List<ui.FontFeature>? fontFeatures;
+
+  @override
+  bool operator ==(Object other) {
```

Comment:
Suggestion: form a signature in constructor so looks can simply compare _signature instead for fast map lookups.

Reply:
Done